bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/Cardio2eBinding.java

```diff
@@ -0,0 +1,944 @@
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.RELAY.name())) {
+					Cardio2eRelayTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for RELAY object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.HVAC_CONTROL.name())) {
+					Cardio2eHvacControlTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for HVAC_CONTROL object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.DATE_AND_TIME.name())) {
+					Cardio2eDateTimeTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for DATE_AND_TIME object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.SCENARIO.name())) {
+					Cardio2eScenarioTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for SCENARIO object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.SECURITY.name())) {
+					Cardio2eSecurityTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for SECURITY object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.ZONES_BYPASS.name())) {
+					Cardio2eZonesBypassTransaction.setSmartSendingEnabledClass (true);
+					logger.debug("Smart sending enabled for ZONES_BYPASS object type in config file");
+				}
+				if (smartSendingEnabledObjectTypesString.contains(Cardio2eObjectTypes.CURTAIN.name())) {
+					Cardio2eCurtainTransaction.setSmartSendingEnabledClass (true);				
+					logger.debug("Smart sending enabled for CURTAIN object type in config file");
+				}
+			}
+	
+			// read further config parameters here ...
+		
+			com = new Cardio2eCom();
+			decoder = new Cardio2eDecoder();
+			receivedDataListener = new ReceivedDataListener();
+			decodedTransactionListener = new DecodedTransactionListener();
+			com.addReceivedDataListener(receivedDataListener);
+			decoder.addDecodedTransactionListener(decodedTransactionListener);
+			com.testMode = testMode;
+			decoder.decodeZonesStateTransaction = zoneStateDetection;
+			com.setSerialPort(port);
+			if (minDelayBetweenReceivingAndSending > 0) com.setMinDelayBetweenReceivingAndSending(minDelayBetweenReceivingAndSending);
+			if (minDelayBetweenSendings > 0) com.setMinDelayBetweenSendings(minDelayBetweenSendings);
+
+			try {
+				com.sendTransaction(new Cardio2eLoginTransaction(programCode)); //Login request
+			} catch (Exception e) {
+				logger.error("Failed to send login request." + e);
+			}
+
+			if (testMode) loggedIn = true;
+			
+			setProperlyConfigured(true);
+			
+			logger.debug("Cardio2e binding activated");
+		} else logger.debug("Cardio2e binding can not be activated because no serial port is set in config file");
```

Comment:
I am sorry. My English teacher told me that the use of contractions is informal. That's why I wrote "can not" instead of "cannot". But if it is wrong, I will change it, of course.